internal/stanza/receiver.go

```diff
@@ -58,23 +59,55 @@ func (r *receiver) Start(ctx context.Context, host component.Host) error {
 			return
 		}
 
```

Comment:
Nit: the entire `func (r *receiver) Start()` func is now a bit too big and with too much nesting. Maybe split to smaller funcs for readability.